src/errors/parser.ts

```diff
@@ -27,6 +27,8 @@ function getCodeByStatus(status: number): string {
         409: ResponseErrorCode.Conflicted,
         429: ResponseErrorCode.TooManyRequests,
         500: ResponseErrorCode.InternalServerError,
+        501: ResponseErrorCode.InternalServerError,
+        502: ResponseErrorCode.Timeout,
```

Comment:
Sorry for that but would it be possible to add the mapping of the checkout? A lot is missing here:

```
const convertProcessingErrorCode = (code: number) => {
    switch (code) {
        case 301:
            return ResponseErrorCode.CardNumberNotValid;

        case 302:
            return ResponseErrorCode.CardExpirationMonthInvalid;

        case 303:
            return ResponseErrorCode.CardExpirationYearInvalid;

        case 304:
            return ResponseErrorCode.CardExpired;

        case 305:
            return ResponseErrorCode.CardCVVInvalid;

        case 306:
            return ResponseErrorCode.CardRejected;

        case 307:
            return ResponseErrorCode.CardInvalid;

        case 308:
            return ResponseErrorCode.ChargeInvalidData;

        case 309:
            return PROCESSING_ERROR;

        case 310:
            return ResponseErrorCode.InvalidUserData;

        case 311:
            return ResponseErrorCode.TooManyChargeRequests;

        case 312:
            return ResponseErrorCode.CancelUnavailable;

        case 313:
            return ResponseErrorCode.ChargeExpired;

        case 314:
            return ResponseErrorCode.SeizeCard;

        case 315:
            return ResponseErrorCode.ContactBank;

        case 316:
            return ResponseErrorCode.LastNameRequired;

        case 317:
            return ResponseErrorCode.PartialCaptureNotSupported;

        case 318:
            return ResponseErrorCode.PartialRefundNotSupported;

        case 319:
            return ResponseErrorCode.FraudSuspected;

        case 320:
            return ResponseErrorCode.BankSystemFailure;

        case 321:
            return ResponseErrorCode.DynamicDescriptorNotSupported;

        case 322:
            return ResponseErrorCode.PaymentCodeInvalid;

        case 323:
            return ResponseErrorCode.PaymentCodeExpired;

        case 324:
            return ResponseErrorCode.PaymentCodeAlreadyUsed;

        case 325:
            return ResponseErrorCode.PaymentCodeStillInUse;

        case 326:
            return ResponseErrorCode.RejectedHighRisk;

        case 327:
            return ResponseErrorCode.ConfirmationPeriodExpired;

        case 328:
            return ResponseErrorCode.RevertFailed;

        case 329:
            return ResponseErrorCode.RefundFailed;

        case 330:
            return ResponseErrorCode.PaymentWalletInsufficientFunds;

        case 331:
            return ResponseErrorCode.InvalidMetadataFieldValue;

        case 332:
            return ResponseErrorCode.CrossBorderNotAcceptedMissingId;

        case 333:
            return ResponseErrorCode.CrossBorderNotAcceptedMissingPhoneNumber;

        case 334:
            return ResponseErrorCode.CrossBorderNotAcceptedUnacceptedPaymentMethod;

        case 335:
            return ResponseErrorCode.CrossBorderNotAcceptedMissingName;

        case 336:
            return ResponseErrorCode.LimitExceededForPaymentType;

        case 337:
            return ResponseErrorCode.LimitExceededForMerchant;

        case 338:
            return ResponseErrorCode.TransactionNotFound;

        case 339:
            return ResponseErrorCode.DuplicateTransaction;

        case 340:
            return ResponseErrorCode.PaymentWalletRejected;

        case 341:
            return ResponseErrorCode.InsufficientMerchantInformation;

        case 342:
            return ResponseErrorCode.CrossBorderNotAcceptedUnacceptedCurrency;

        case 343:
            return ResponseErrorCode.GatewayServerError;

        case 344:
            return ResponseErrorCode.PaymentMethodTemporarilyUnavailable;

        case 345:
            return ResponseErrorCode.PaymentCanceled;

        case 346:
            return ResponseErrorCode.ExceededPendingThreshold;

        default:
            return null;
    }
};
```

No need to include the default and `PROCESSING_ERROR`. Then we can just use this on the checkout directly 🙂

Reply:
Sure! I'll add them :muscle:

Reply:
Thanks a lot 🙂